crates/aranya-client/src/aqc/net.rs

```diff
@@ -0,0 +1,373 @@
+#![warn(missing_docs)]
+
+//! The AQC network implementation.
+
+use std::{
+    collections::HashMap,
+    net::{Ipv4Addr, SocketAddr},
+    sync::Arc,
+    task::{Context, Poll, Waker},
+};
+
+use aranya_crypto::aqc::{BidiChannelId, UniChannelId};
+use aranya_daemon_api::{
+    AqcBidiPsks, AqcCtrl, AqcPsks, AqcUniPsks, DaemonApiClient, LabelId, TeamId,
+};
+use buggy::{Bug, BugExt as _};
+use bytes::Bytes;
+use channels::AqcPeerChannel;
+use s2n_quic::{self, client::Connect, provider, Client, Connection, Server};
+use tarpc::context;
+use tokio::sync::mpsc;
+use tracing::{debug, error, warn};
+
+use super::{
+    api::AqcChannelId,
+    crypto::{ClientPresharedKeys, ServerPresharedKeys, CTRL_KEY, PSK_IDENTITY_CTRL},
+};
+use crate::error::{aranya_error, AqcError, IpcError};
+
+pub mod channels;
+
+/// An AQC client. Used to create and receive channels.
+#[derive(Debug)]
+pub(crate) struct AqcClient {
+    quic_client: Client,
+    client_keys: Arc<ClientPresharedKeys>,
+    server_keys: Arc<ServerPresharedKeys>,
+    /// Map of PSK identity to channel type
+    channels: HashMap<Vec<u8>, AqcChannelInfo>,
```

Comment:
We should define a type alias for the PSK identity for readability:
```suggestion
    channels: HashMap<PskId, AqcChannelInfo>,
```